shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/ingest/wal/WalEventConverter.java

```diff
@@ -125,9 +125,9 @@ private DataRecord createDataRecord(final AbstractRowEvent rowsEvent, final int
     
     private void putColumnsIntoDataRecord(final DataRecord dataRecord, final PipelineTableMetaData tableMetaData, final List<Object> values) {
         for (int i = 0, count = values.size(); i < count; i++) {
-            boolean isUniqueKey = tableMetaData.isUniqueKey(i);
+            boolean isUniqueKey = tableMetaData.isUniqueKey(i + 1);
             Object uniqueKeyOldValue = isUniqueKey ? values.get(i) : null;
-            Column column = new Column(tableMetaData.getColumnMetaData(i).getName(), uniqueKeyOldValue, values.get(i), true, isUniqueKey);
+            Column column = new Column(tableMetaData.getColumnMetaData(i + 1).getName(), uniqueKeyOldValue, values.get(i), true, isUniqueKey);
```

Comment:
It could be simplified after move `uniqueKey` to `PipelineColumnMetaData`. Get column metadata firstly, then check whether it's unique key or not.